### PR TITLE
fix: fix kms reference error when sm crn is set

### PR DESCRIPTION
**Pull request by vburckhardt**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Review by akocbek — DISMISSED** [review]

File: solutions/standard/outputs.tf [hunk 1]
```diff
@@ -14,12 +14,12 @@ output "secrets_manager_guid" {
 }
 
 output "secrets_manager_id" {
-  description = "ID of Secrets Manager instance"
+  description = "ID of Secrets Manager instance. Same value as secrets_manager_guid"
   value       = var.existing_secrets_manager_crn == null ? module.secrets_manager[0].secrets_manager_id : local.secrets_manager_guid
```
Comment (akocbek) [inline solutions/standard/outputs.tf:18]: data lookup shows the following
```
 crn                     = "crn:v1:bluemix:public:secrets-manager:us-south:a/190c293e9fda4c6684b5acf4b17871b8:6bf6666d-73a6-41ad-a9ae-86cf9c9953e0::"
      + extensions              = {
          + "virtual_private_endpoints.dns_domain"             = "private.us-south.secrets-manager.appdomain.cloud"
          + "virtual_private_endpoints.dns_hosts.#"            = "1"
          + "virtual_private_endpoints.dns_hosts.0"            = "6bf6666d-73a6-41ad-a9ae-86cf9c9953e0"
          + "virtual_private_endpoints.endpoints.#"            = "3"
          + "virtual_private_endpoints.endpoints.0.ip_address" = "166.9.228.16"
          + "virtual_private_endpoints.endpoints.0.zone"       = "dal10"
          + "virtual_private_endpoints.endpoints.1.ip_address" = "166.9.229.17"
          + "virtual_private_endpoints.endpoints.1.zone"       = "dal12"
          + "virtual_private_endpoints.endpoints.2.ip_address" = "166.9.230.13"
          + "virtual_private_endpoints.endpoints.2.zone"       = "dal13"
          + "virtual_private_endpoints.origin_type"            = "cse"
          + "virtual_private_endpoints.ports.#"                = "1"
          + "virtual_private_endpoints.ports.0.port_max"       = "443"
          + "virtual_private_endpoints.ports.0.port_min"       = "443"
        }
      + guid                    = "6bf6666d-73a6-41ad-a9ae-86cf9c9953e0"
      + id                      = "crn:v1:bluemix:public:secrets-manager:us-south:a/190c293e9fda4c6684b5acf4b17871b8:6bf6666d-73a6-41ad-a9ae-86cf9c9953e0::"
      + identifier              = "crn:v1:bluemix:public:secrets-manager:us-south:a/190c293e9fda4c6684b5acf4b17871b8:6bf6666d-73a6-41ad-a9ae-86cf9c9953e0::"
      + location                = "us-south"
```
id is not similar to guid. Is that just in that case?

maybe we could use ?
`value       = var.existing_secrets_manager_crn == null ? module.secrets_manager[0].secrets_manager_id : data.ibm_resource_instance.existing_sm[0].id`

File: solutions/standard/variables.tf [hunk 2]
```diff
@@ -16,7 +16,8 @@ variable "use_existing_resource_group" {
 
 variable "resource_group_name" {
   type        = string
-  description = "The name of a new or an existing resource group in which to provision Secrets Manager resources to. If prefix input variable is passed then it will get prefixed infront of the value in the format of '<prefix>-value'"
+  description = "The name of a new or an existing resource group in which to provision Secrets Manager resources to. If prefix input variable is passed then it will get prefixed infront of the value in the format of '<prefix>-value'. Optional if existing_secrets_manager_crn is set."
```
Comment (akocbek) [inline solutions/standard/variables.tf:19]: shouldn't be here if `existing_secrets_manager_crn` is **not** set?  
according to `count = var.existing_secrets_manager_crn == null ? 1 : 0` we do not  create resource group if `existing_secrets_manager_crn` is set

File: solutions/standard/outputs.tf (hunk 1 above)
Comment (vburckhardt) [reply solutions/standard/outputs.tf:18]: Correct, but it is in the module output, isn't it? This is why I added this in the description.

Comment (vburckhardt) [reply solutions/standard/outputs.tf:18]: Let's track it separately. I need to get this on merged as functionality missing is a blocker

**Review by akocbek — APPROVED** [review]

**Pull request merged**